services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -73,9 +81,53 @@ async def collect_garbage(registry: RedisResourceRegistry, app: web.Application)
                 )
 
                 await remove_resources_if_guest_user(
-                    app=app, project_uuid=resource_value, user_id=int(key["user_id"])
+                    app=app,
+                    project_uuid=resource_value,
+                    user_id=int(dead_key["user_id"]),
                 )
 
+    # remove possible pending contianers
+    await remove_orphaned_services(registry, app)
+
+
+async def remove_orphaned_services(
+    registry: RedisResourceRegistry, app: web.Application
+) -> None:
+    """Removes services which are no longer tracked in the database
+
+    Multiple deployments can be active at the same tim on the same cluster.
+    This will also check the current SWARM_STACK_NAME of the service which
+    must be matching its own.
+
+    If the service is a dynamic service
+    """
+    logger.info("Starting orphaned services removal...")
+    currently_opened_projects = set()
+    alive_keys, _ = await registry.get_all_resource_keys()
+    for alive_key in alive_keys:
+        resources = await registry.get_resources(alive_key)
+        if "project_id" not in resources:
+            continue
+
+        node_id = await get_node_id_from_project_id(app, resources["project_id"])
+        currently_opened_projects.update(node_id)
+
+    running_interactive_services = await get_running_interactive_services(app)
+    logger.info(
+        "Will collect the following: %s",
+        [x["service_host"] for x in running_interactive_services],
+    )
+    for interactive_service in running_interactive_services:
+        # if not present in DB or not part of currently opened projects, can be removed
+        if (
+            not await is_service_present_in_db(app, interactive_service["service_uuid"])
+            or interactive_service["service_uuid"] not in currently_opened_projects
+        ):
+            logger.info("Will remove service %s", interactive_service["service_host"])
+            await stop_service(app, interactive_service["service_uuid"])
```

Comment:
Since we do not care about the result nor to parallelise the stop. I would prefer to do it sequentially. Also speed is not an issue for the garbage collector.